From 95450a7ae8aff926c4d78970ca33c223894d34c5 Mon Sep 17 00:00:00 2001
From: Jani Hautakangas <jani@kodegood.com>
Date: Sun, 24 Aug 2025 00:18:34 +0300
Subject: [PATCH 3/7] Build Lagom tools when cross compiling

---
 Meta/Lagom/CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Meta/Lagom/CMakeLists.txt b/Meta/Lagom/CMakeLists.txt
index cd77177649..a3c5419219 100644
--- a/Meta/Lagom/CMakeLists.txt
+++ b/Meta/Lagom/CMakeLists.txt
@@ -73,10 +73,10 @@ endif()
 # Instead, we import them from a previous install of Lagom. This mandates a two-stage build for fuzzers.
 # The same concern goes for cross-compile builds, where we need the tools built for the host
 set(BUILD_LAGOM_TOOLS ON)
-if (ENABLE_FUZZERS OR CMAKE_CROSSCOMPILING)
-    find_package(LagomTools REQUIRED)
-    set(BUILD_LAGOM_TOOLS OFF)
-endif()
+# if (ENABLE_FUZZERS OR CMAKE_CROSSCOMPILING)
+#     find_package(LagomTools REQUIRED)
+#     set(BUILD_LAGOM_TOOLS OFF)
+# endif()
 
 include(lagom_compile_options)
 
-- 
2.48.1

